### ISSUE TITLE
Edge core not getting Ready

### ISSUE THREAD

**Issue by sharmmoh1983** [kubeedge/kubeedge#636]
Edge node (raspberry pi)was initially  getting connected to master node(ec2) but now it is now in not ready state 

Logs:
```
ERROR wsserver/server.go:147 failed to consume event for node raspi, reason: failed to get message from cluster, reason: channel is closed
```

what can be the root cause?

```
controller/upstream.go:97 Parse message: f2922c5e-ee5b-4a58-b12c-d72ed9c1b707 resource type with error: unknown resource
```

due to which edge core is not getting connected

**Comment by rohitsardesai83** [comment]
@sharmmoh1983 , please check the edge core log , if the web socket connection to the cloud is ok.

**Comment by sharmmoh1983** [comment]
@rohitsardesai83 I have checked the logs and websocket connection to cloud is fine

**Comment by rohitsardesai83** [comment]
@naveensriram , can we ask someone to test this issue once ? 

@sharmmoh1983 , please try stopping edge_core , clean the edge.db on the rpi , and then restart the edge_core process.  Please collect edge_core logs and share them once you observe the issue again.

**Comment by subpathdev** [comment]
@sharmmoh1983 can you please upload your edge/conf/edge.yaml?

**Comment by sharmmoh1983** [comment]
@rohitsardesai83  I have created new edge core on raspi still same issue 

Logs:
2019-06-08 04:05:55.465 +00:00 ERROR edged/edged_status.go:354 Unable to construct v1.Node object for edge: operation timeout: context deadline exceeded

2019-06-08 04:05:55.306 +00:00 INFO metamanager/msg_processor.go:304 list pod status, no record, skip sync
2019-06-08 04:05:55.313 +00:00 ERROR containers/container_manager.go:510 list container error operation timeout: context deadline exceeded
2019-06-08 04:05:55.320 +00:00 INFO dtcontext/dtcontext.go:83 MemModule is healthy 1559966755
2019-06-08 04:05:55.320 +00:00 INFO dtcontext/dtcontext.go:83 TwinModule is healthy 1559966755
2019-06-08 04:05:55.320 +00:00 INFO dtcontext/dtcontext.go:83 DeviceModule is healthy 1559966755
2019-06-08 04:05:55.320 +00:00 INFO dtcontext/dtcontext.go:83 CommModule is healthy 1559966755
2019-06-08 04:05:55.366 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:05:55.465 +00:00 ERROR edged/edged_status.go:354 Unable to construct v1.Node object for edge: operation timeout: context deadline exceeded
2019-06-08 04:05:56.366 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:05:57.367 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:05:58.367 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:05:59.368 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:00.368 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:01.369 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:02.369 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:03.369 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:04.370 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:05.313 +00:00 INFO metamanager/msg_processor.go:416 get a message {Header:{ID:eac19f31-b270-479b-8881-9ab674602605 ParentID: Timestamp:1559966765313 Sync:false} Router:{Source:edged Group:edged Operation:query Resource:default/pod} Content:<nil>}
2019-06-08 04:06:05.315 +00:00 INFO edged/edged.go:612 request content is null
2019-06-08 04:06:05.370 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:06.371 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:07.371 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:08.372 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:09.372 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:10.373 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:11.373 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:12.373 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:13.374 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:14.374 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:15.375 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:16.375 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:17.376 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:18.376 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:19.377 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:20.377 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:21.377 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:22.378 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:23.378 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:24.379 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:25.379 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:26.380 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:27.380 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:28.381 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:29.381 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:30.381 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:31.382 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:32.382 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:33.383 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:34.383 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:35.383 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:36.384 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:37.384 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:38.385 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:39.385 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:40.386 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:41.386 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:42.387 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:43.387 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:44.387 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:45.388 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:46.389 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:47.389 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:48.390 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:49.390 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:50.391 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:51.391 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:52.391 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:53.392 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:54.392 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:55.296 +00:00 INFO metamanager/msg_processor.go:416 get a message {Header:{ID:9bc6c3b4-cfad-448b-929c-b36fee7f3d5b ParentID: Timestamp:1559966815296 Sync:false} Router:{Source:metaManager Group:resource Operation:meta-internal-sync Resource:podstatus} Content:<nil>}
2019-06-08 04:06:55.297 +00:00 INFO metamanager/msg_processor.go:297 start to sync pod status
2019-06-08 04:06:55.298 +00:00 INFO metamanager/msg_processor.go:304 list pod status, no record, skip sync
2019-06-08 04:06:55.320 +00:00 INFO dtcontext/dtcontext.go:83 TwinModule is healthy 1559966815
2019-06-08 04:06:55.320 +00:00 INFO dtcontext/dtcontext.go:83 CommModule is healthy 1559966815
2019-06-08 04:06:55.320 +00:00 INFO dtcontext/dtcontext.go:83 MemModule is healthy 1559966815
2019-06-08 04:06:55.320 +00:00 INFO dtcontext/dtcontext.go:83 DeviceModule is healthy 1559966815
2019-06-08 04:06:55.393 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:56.393 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:57.394 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:58.394 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:06:59.395 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:00.395 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:01.395 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:02.396 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:03.396 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:04.397 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:05.397 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:06.398 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:07.398 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:08.398 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:09.399 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:10.399 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:11.400 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:12.400 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:13.401 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:14.401 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:15.402 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:16.402 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:17.403 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:18.403 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:19.404 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:20.404 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:21.405 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:22.405 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:23.406 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:24.406 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:25.407 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:26.407 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:27.407 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:28.408 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:29.408 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:30.409 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:31.409 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:32.409 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:33.410 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:34.410 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:35.411 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:36.411 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:37.411 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:38.412 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:39.412 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:40.413 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:41.413 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:42.414 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:43.414 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:44.414 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:45.415 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:46.415 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:47.416 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:48.416 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:49.417 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:50.417 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:51.418 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:52.418 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
^A2019-06-08 04:07:53.419 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:54.420 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:55.296 +00:00 INFO metamanager/msg_processor.go:416 get a message {Header:{ID:9f125cca-073f-4597-a833-5725ee0ad362 ParentID: Timestamp:1559966875296 Sync:false} Router:{Source:metaManager Group:resource Operation:meta-internal-sync Resource:podstatus} Content:<nil>}
2019-06-08 04:07:55.297 +00:00 INFO metamanager/msg_processor.go:297 start to sync pod status
2019-06-08 04:07:55.297 +00:00 INFO metamanager/msg_processor.go:304 list pod status, no record, skip sync
2019-06-08 04:07:55.320 +00:00 INFO dtcontext/dtcontext.go:83 CommModule is healthy 1559966875
2019-06-08 04:07:55.320 +00:00 INFO dtcontext/dtcontext.go:83 DeviceModule is healthy 1559966875
2019-06-08 04:07:55.320 +00:00 INFO dtcontext/dtcontext.go:83 MemModule is healthy 1559966875
2019-06-08 04:07:55.320 +00:00 INFO dtcontext/dtcontext.go:83 TwinModule is healthy 1559966875
2019-06-08 04:07:55.420 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:55.690 +00:00 ERROR edged/edged_status.go:354 Unable to construct v1.Node object for edge: operation timeout: context deadline exceeded
2019-06-08 04:07:56.421 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:57.421 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting
2019-06-08 04:07:58.422 +00:00 INFO pleg/generic.go:47 GenericLifecycle: Relisting

**Comment by sharmmoh1983** [comment]
@subpathdev 

Please find the following edge.yaml
```
mqtt:
    server: tcp://127.0.0.1:1883 # external mqtt broker url.
    internal-server: tcp://127.0.0.1:1884 # internal mqtt broker url.
    mode: 0 # 0: internal mqtt broker enable only. 1: internal and external mqtt broker enable. 2: external mqtt broker enable only.
    qos: 0 # 0: QOSAtMostOnce, 1: QOSAtLeastOnce, 2: QOSExactlyOnce.
    retain: false # if the flag set true, server will store the message and can be delivered to future subscribers.
    session-queue-size: 100 # A size of how many sessions will be handled. default to 100.

edgehub:
    websocket:
        url: wss://ec2-13-236-186-114.ap-southeast-2.compute.amazonaws.com:10000/e632aba927ea4ac2b575ec1603d56f10/raspi/events
        certfile: /tmp/edge.crt
        keyfile: /tmp/edge.key
        handshake-timeout: 30 #second
        write-deadline: 15 # second
        read-deadline: 15 # second
    controller:
        placement: false
        heartbeat: 15  # second
        refresh-ak-sk-interval: 10 # minute
        auth-info-files-path: /var/IEF/secret
        placement-url: https://10.154.193.32:7444/v1/placement_external/message_queue
        project-id: e632aba927ea4ac2b575ec1603d56f10
        node-id: raspi

edged:
    register-node-namespace: default
    hostname-override: raspi
    interface-name: eth0
    node-status-update-frequency: 10 # second
    device-plugin-enabled: false
    gpu-plugin-enabled: false
    image-gc-high-threshold: 80 # percent
    image-gc-low-threshold: 40 # percent
    maximum-dead-containers-per-container: 1
    version: 2.0.0
```

**Comment by sharmmoh1983** [comment]
@rohitsardesai83  Can I use Kubeedge installer to install edge core on raspi? Any steps for the same?

**Comment by subpathdev** [comment]
You can not use an url to connect to the cloud part. (compare to pr #645 )

**Comment by sharmmoh1983** [comment]
@subpathdev I have been using the same configuration earlier and edge node was getting connected but now it is working so I don't think url is the issue

**Comment by rohitsardesai83** [comment]
> @rohitsardesai83 Can I use Kubeedge installer to install edge core on raspi? Any steps for the same?

No the installer doesn't support arm.

**Comment by rohitsardesai83** [comment]
> 2019-06-08 04:05:55.465 +00:00 **ERROR** edged/edged_status.go:354 Unable to construct v1.Node object for edge: operation timeout: context deadline exceeded

2019-06-08 04:05:55.465 +00:00 **ERROR** edged/edged_status.go:354 Unable to construct v1.Node object for edge: operation timeout: context deadline exceeded

We need to debug further why is this error coming .
Can you put more logs in the intialNode func 

https://github.com/kubeedge/kubeedge/blob/master/edge/pkg/edged/edged_status.go#L352

**Comment by sharmmoh1983** [comment]
@rohitsardesai83 There is no source code available in raspi installable .There is only executable file so can't put log statements in code

Also following is appearing in edgecontroller :
2019-06-09 06:13:36.339 +00:00 ERROR handler/websockethandler.go:89 read error, connection for node raspi will be closed, reason: read tcp 172.31.5.21:10000->49.195.101.134:20546: i/o timeout
2019-06-09 06:13:36.339 +00:00 ERROR handler/eventhandler.go:238 failed to consume event for node raspi, reason: failed to get message from cluster, reason: channel is closed

**Comment by SupriyaSupu** [comment]
@rohitsardesai83 @sharmmoh1983 @naveensriram 

I tried to regenerate the scenario by connecting Edge node which is running on raspberry pi to master node which is running on Ubuntu OS on laptop. Its working fine. I tried to stop edge_core and deleted node and tried again to connect. I have checked log as well.  In all cases its working fine.

I am not able to reproduce the error.

**Comment by sharmmoh1983** [comment]
@SupriyaSupu 

Can you provide me list of steps you followed right from start (from cloud to edge)  installation and making them connected and get in  ready state, so that I can follow the same and see the difference?

**Comment by rohitsardesai83** [comment]
> @rohitsardesai83 @sharmmoh1983 @naveensriram
> 
> I tried to regenerate the scenario by connecting Edge node which is running on raspberry pi to master node which is running on Ubuntu OS on laptop. Its working fine. I tried to stop edge_core and deleted node and tried again to connect. I have checked log as well. In all cases its working fine.
> 
> I am not able to reproduce the error.

@SupriyaSupu , please try to recreate with cloud vm instead of laptop.

**Comment by SupriyaSupu** [comment]
@sharmmoh1983 

Cloud: 
Follows steps from link given below to run cloud:
https://github.com/kubeedge/kubeedge/blob/master/docs/getting-started/release_package.md

Edge: 
1. Download binary to rasberry pi from:
https://github.com/kubeedge/kubeedge/releases
named "kubeedge-v0.3.0-linux-arm.tar.gz"

2. Untar binary and change master IP with your VM IP in kubeedge/edge/conf/edge.yaml file

3. Copy certificate to raspberry pi. Update the path of certificates in kubeedge/edge/conf/edge.yaml file.

4. Run the edge_core on raspberry pi.

**Comment by SupriyaSupu** [comment]
@rohitsardesai83 
I will try to recreate with cloud VM.

**Comment by SupriyaSupu** [comment]
@sharmmoh1983 @rohitsardesai83 
I tried to reproduce scenario with cloud VM. I am not able to reproduce the error.

**Comment by sids-b** [comment]
@sharmmoh1983 , are you still facing this issue ??

**Comment by sids-b** [comment]
closing this issue. please free to reach out us via slack or reopen this issue for further discussions :)

**Comment by qq897807460** [comment]
Please help me!!!
In the cloud, we encountered the same problem. At the edge, it seems that there is no connectivity.
Cloud：
         ERROR wsserver/server.go:147 failed to consume event for node raspi, reason: failed to get message from cluster, reason: channel is closed
Edge：
         
![edge](https://user-images.githubusercontent.com/44339024/61619714-fadcd780-aca1-11e9-8e93-1b52e515f9ce.jpg)

we have modified the $GOPATH/src/github.com/kubeedge/kubeedge/edge/conf/edge.yaml configuration file.

**Comment by sids-b** [comment]
@sharmmoh1983 , are you trying with the latest master source code ?

**Comment by sjw1230** [comment]
> @rohitsardesai83 There is no source code available in raspi installable .There is only executable file so can't put log statements in code
> 
> Also following is appearing in edgecontroller :
> 2019-06-09 06:13:36.339 +00:00 ERROR handler/websockethandler.go:89 read error, connection for node raspi will be closed, reason: read tcp 172.31.5.21:10000->49.195.101.134:20546: i/o timeout
> 2019-06-09 06:13:36.339 +00:00 ERROR handler/eventhandler.go:238 failed to consume event for node raspi, reason: failed to get message from cluster, reason: channel is closed

we encountered the same problem， how do you solve it？

**Comment by bitvijays** [comment]
@sjw1230 Were you able to solve this issue?
